ckanext/ddi/tests/test_blueprints.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+
+import os
+import pytest
+import six
+import ckan.plugins.toolkit as toolkit
+import ckan.lib.uploader
+from ckantoolkit.tests import factories
+
+
+def _assert_in_body(string, response):
+    if six.PY2:
+        assert string in response.body.decode('utf8')
+    else:
+        assert string in response.body
+
+
+def _post_request(app, url, form, files, environ, **kwargs):
+    try:
+        # CKAN 2.9
+        data = dict(form)
+        for field, filename in files.items():
+            data[field] = _load_test_data(filename)
+        return app.post(
+            url, data=data, environ_overrides=environ, follow_redirects=False, **kwargs
+        )
+    except TypeError:
+        # CKAN 2.8
+        return app.post(
+            url,
+            params=form,
+            upload_files=[
+                (field, filename, _load_test_data(filename).read(),)
+                for field, filename in files.items()
+            ],
+            extra_environ=environ,
+            **kwargs
+        )
+
+
+def _load_test_data(filename):
+    return open(os.path.join(os.path.dirname(__file__), 'test_data', filename), 'rb')
+
+
+def _patch_storage_path(monkeypatch, tmpdir, ckan_config):
```

Comment:
Unforuntately I wasn't able to do this in `setup()` because it doesn't get passed the vars `monkeypatch`, `tmpdir` and `ckan_config` so I am just calling `_patch_storage_path(monkeypatch, tmpdir, ckan_config)` at the start of every test that needs it. Can you think of any way round this that would allow us to do this in `setup()`?

Reply:
What about creating your own `patch_storage` fixture that does this and applying it to the class alongside `clean_db` etc?

Reply:
OK. I think I'm going to punt this job to an issue